R/train.R

```diff
@@ -31,6 +31,10 @@
 train = function(learner, task, subset, weights = NULL) {
   learner = checkLearner(learner)
   assertClass(task, classes = "Task")
+  if (hasExpression(learner)) {
+    dict = getTaskDictionary(task = task)
```

Comment:
Subsetting happens in `trainLearner` ([e.g.](https://github.com/mlr-org/mlr/blob/c2d40a3264ef564391260e38acf5be4d8c4f7473/R/RLearner_classif_logreg.R#L22))